lib/src/chart/bar_chart/bar_chart_painter.dart

```diff
@@ -701,6 +701,12 @@ class BarChartPainter extends AxisChartPainter<BarChartData> {
         bottomRight: radius);
     _bgTouchTooltipPaint.color = tooltipData.tooltipBgColor;
 
+    /// if tooltip gradient is not null then show gradient instead of plain color
+    if (tooltipData.tooltipBgGradient != null) {
```

Comment:
As we reuse this `_bgTouchTooltipPaint`, please write both if and else branches and make `_bgTouchTooltipPaint.shader = null` when we have color, otherwise make `bgTouchTooltipPaint.color = null` or `Colors.transparent`.